### DISCUSSION TITLE
How do you set the length of summary excerpt text

### DISCUSSION THREAD

**Discussion by nogginbox** [OrchardCMS/OrchardCore#10428] [Q&A] [answered]
When body text is displayed in Orchard Core a certain number of characters or words is displayed.

How do you set the length? I can't find it in any of the templates or in the admin section.

**Comment by deanmarcussen** [comment, marked as answer]
The string extension is in the namepsace `using OrchardCore.ContentManagement.Utilities` and is called `Ellipsize`

**Reply by hishamco** [reply]
Also you can use `truncate` & `truncatewords` if you are using Liquid

**Reply by nogginbox** [reply]
Great. I've been converting an Orchard Classic theme and keep looking for theme templates and methods that don't exist anymore.

In this case I needed to override **HtmlBodyPart.Summary.cshtml** with:
```
@using OrchardCore.Mvc.Utilities
@model OrchardCore.Html.ViewModels.HtmlBodyPartViewModel
<p>@Html.Raw(Model.Html.RemoveTags(true).Ellipsize(600).ReplaceNewLinesWith($"</p>{Environment.NewLine}<p>"))</p>
```

**Comment by pgy866** [comment]
https://github.com/OrchardCMS/OrchardCore/issues/2889

**Reply by hishamco** [reply]
@pgy866 how this related? I think @NogginBox means display a truncate text based on user-define length

**Reply by pgy866** [reply]
> @pgy866 how this related? I think @NogginBox means display a truncate text based on user-define length

More than just the interception of "summary excerpt text"， It is supported to add a manual setting to limit its length when creating a field. For example, in certain situations, limiting its length, would it be great?

**Reply by nogginbox** [reply]
I can see why setting field length would be a useful feature, particularly if there were rules imposed by the business on this. But, in my case it was not what I was looking for.
